server/public/model/client4_test.go

```diff
@@ -105,3 +106,46 @@ func TestClient4SetToken(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Equal(t, http.StatusOK, resp.StatusCode)
 }
+
+func TestClient4RequestCancellation(t *testing.T) {
+	t.Run("cancel before making the reqeust", func(t *testing.T) {
+		server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			t.Fatal("request should not hit the server")
+		}))
+
+		client := NewAPIv4Client(server.URL)
+
+		ctx, cancel := context.WithCancel(context.Background())
+
+		cancel()
+
+		_, resp, err := client.GetMe(ctx, "")
```

Comment:
In both tests, can we verify that the error is `context.Canceled`?

Reply:
Good point, added in 4cbc745f42.